### PR TITLE
Document MDC log correlation support

### PR DESCRIPTION
**Pull request by avimas**
## Which problem is this PR solving?
documentation for #718

## Short description of the changes
add documentation for log correlation

### CODE REVIEW

File: jaeger-core/README.md [hunk 1]
```diff
@@ -140,6 +140,36 @@ Tracer tracer = configuration
...
     .build();
 ```
+### Log Correlation
+
+The Jaeger Java Client also provides log correlation support by configuring the logging context (MDC)
+with the following variables:
+  
+trace id - %{traceId}  
+span id  - %{spanId}  
+sampled  - %{sampled}
```
Comment (yurishkuro) [inline jaeger-core/README.md:150]:
```suggestion
The Jaeger Java Client supports log correlation by updating the logging context (MDC) with three variables: `traceId`, `spanId`, `sampled`.
```

File: jaeger-core/README.md [hunk 2]
```diff
@@ -140,6 +140,36 @@ Tracer tracer = configuration
...
+The Jaeger Java Client also provides log correlation support by configuring the logging context (MDC)
+with the following variables:
+  
+trace id - %{traceId}  
+span id  - %{spanId}  
+sampled  - %{sampled}  
+
+To accomplish that Jaegar Tracer should be created with the following two steps:
+
+1. Create the MDCScopeManager using either default names:
```
Comment (yurishkuro) [inline jaeger-core/README.md:154]:
```suggestion
1. Create the MDCScopeManager using either default MDC field names:
```

File: jaeger-core/README.md [hunk 3]
```diff
@@ -140,6 +140,36 @@ Tracer tracer = configuration
...
+                                   .Builder()
+                                   .withMDCTraceIdKey("CustomTraceId")
+                                   .withMDCSampledKey("customSampled")
+                                   .withMDCSpanIdKey("customSpanId")
+                                   .build();
+    ```
+2. Create the Jaegar Tracer by supplying the MDCScopeManager created step 1:
+```java
+JaegerTracer.Builder("serviceName").withScopeManager(scopeManager).build();	
+```
```
Comment (yurishkuro) [inline jaeger-core/README.md:172]: I would also add that the log formatter must be configured to include MDC fields in the output. Show example of the formatter for, say, log4j, and include a sample output log line.

Comment (avimas) [reply jaeger-core/README.md:168]: Thanks for the feedback!

**Pull request merged**